apps/src/codebridge/utils/getUrlForFile.ts

```diff
@@ -0,0 +1,19 @@
+import {ProjectFile} from '../types';
+
+// Given a ProjectFile, either create a Blob URL for it or get its external URL.
+export const getUrlForFile = (
+  file: ProjectFile,
+  parentOrigin: string,
+  externalFileTypes: string[]
+): string => {
+  let fileType = '';
+  if (externalFileTypes.includes(file.language)) {
```

Comment:
You mentioned in description that `weblab` encodes images as blobs. Out of curiosity, do you mind pointing where this happens?